src/Algorand/AssetTransaction.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "AssetTransaction.h"
+#include "BinaryCoding.h"
+
+using namespace TW;
+using namespace TW::Algorand;
+
+Data AssetTransaction::serialize() const {
+    Data data;
+
+    // encode map length
+    uint8_t size = 10;
+    if (!note.empty()) {
+        // note is optional
+        size += 1;
+    }
+    data.push_back(0x80 + size);
+
+    // encode fields one by one (sorted by name)
+    encodeString("aamt", data);
+    encodeNumber(amount, data);
+
+    encodeString("arcv", data);
+    encodeBytes(Data(to.bytes.begin(), to.bytes.end()), data);
+
+    encodeString("fee", data);
+    encodeNumber(fee, data);
+
+    encodeString("fv", data);
+    encodeNumber(firstRound, data);
+
+    encodeString("gen", data);
+    encodeString(genesisId, data);
+
+    encodeString("gh", data);
+    encodeBytes(genesisHash, data);
+
+    encodeString("lv", data);
+    encodeNumber(lastRound, data);
+
+    if (!note.empty()) {
+        encodeString("note", data);
+        encodeBytes(note, data);
+    }
+
+    encodeString("snd", data);
+    encodeBytes(Data(from.bytes.begin(), from.bytes.end()), data);
+
+    encodeString("type", data);
+    encodeString(type, data);
+
+    encodeString("xaid", data);
+    encodeNumber(assetId, data);
+
+    return data;
+}
+
+Data AssetTransaction::serialize(Data& signature) const {
+    return BaseTransaction::serialize(signature);
+}
```

Comment:
Since this method just invokes base class method, it can be omitted.

```suggestion
```

Reply:
I can't remove it. It's not compiled. I can move it in header, is it ok? Or give me an advice how to fix it.

`Undefined symbols for architecture x86_64:
  "TW::Algorand::Transaction::serialize(std::__1::vector<unsigned char, std::__1::allocator<unsigned char> > const&) const", referenced from:
      AlgorandSigner_Sign_Test::TestBody() in SignerTests.cpp.o
      TW::Algorand::Signer::sign(TW::Algorand::Proto::SigningInput const&) in libTrustWalletCore.a(Signer.cpp.o)
      vtable for TW::Algorand::Transaction in libTrustWalletCore.a(Transaction.cpp.o)
  "TW::Algorand::AssetTransaction::serialize(std::__1::vector<unsigned char, std::__1::allocator<unsigned char> > const&) const", referenced from:
      vtable for TW::Algorand::AssetTransaction in libTrustWalletCore.a(AssetTransaction.cpp.o)`

Reply:
Or i can do like `auto result = transaction.BaseTransaction::serialize(signature);`

Reply:
move to header file as default implementation

Reply:
maybe you can mark the base implementation virtual (now only the specialized ones are marked override)